Source/Microsoft.Teams.Apps.FAQPlusPlus.Configuration/Controllers/HomeController.cs

```diff
@@ -228,27 +223,20 @@ public async Task<string> GetSavedHelpTabTextAsync()
         /// Based on deep link URL received find team id and return it to that it can be saved.
         /// </summary>
         /// <param name="teamIdDeepLink">team id deep link.</param>
-        /// <returns>team id as string.</returns>
+        /// <returns>team id decoded string.</returns>
         private static string ParseTeamIdFromDeepLink(string teamIdDeepLink)
         {
-            int startEscapedIndex = teamIdDeepLink.IndexOf(TeamIdEscapedStartString, StringComparison.OrdinalIgnoreCase);
-            int endEscapedIndex = teamIdDeepLink.IndexOf(TeamIdEscapedEndString, StringComparison.OrdinalIgnoreCase);
-
-            int startUnescapedIndex = teamIdDeepLink.IndexOf(TeamIdUnescapedStartString, StringComparison.OrdinalIgnoreCase);
-            int endUnescapedIndex = teamIdDeepLink.IndexOf(TeamIdUnescapedEndString, StringComparison.OrdinalIgnoreCase);
-
-            string teamID = string.Empty;
+            // team id regex match
+            // for a pattern like https://teams.microsoft.com/l/team/19%3a64c719819fb1412db8a28fd4a30b581a%40thread.tacv2/conversations?groupId=53b4782c-7c98-4449-993a-441870d10af9&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47
+            // regex checks for 19%3a64c719819fb1412db8a28fd4a30b581a%40thread.tacv2
+            var match = Regex.Match(teamIdDeepLink, @"/team\/([\s\S %?]*)\/");
```

Comment:
[\s\S %?] doesn't make sense as a character group. \s and \S are complements, so that's effectively *

Reply:
Change this to @"teams.microsoft.com/l/team/(\S+)/"